smarts/core/sensors.py

```diff
@@ -522,6 +520,18 @@ def _vehicle_is_wrong_way(sim, vehicle, lane_id):
             > 0.5 * np.pi
         )
 
+    @classmethod
+    def _check_wrong_way_event(cls, lane_to_check, sim, vehicle):
+        # When the vehicle is in an intersection, turn off the `wrong way` check to avoid
+        # false positive `wrong way` events.
+        if lane_to_check.getEdge().isSpecial():
+            is_wrong_way = False
+        else:
+            is_wrong_way = cls._vehicle_is_wrong_way(
+                sim, vehicle, lane_to_check.getID()
+            )
+        return is_wrong_way
```

Comment:
Not that big of a deal, but this can be simplified to:
```python
@classmethod
def _check_wrong_way_event(cls, lane_to_check, sim, vehicle):
    # When the vehicle is in an intersection, turn off the `wrong way` check to avoid
    # false positives 'wrong way' events.  (see Issue #846)
    if lane_to_check.getEdge().isSpecial():
        return False
    return cls._vehicle_is_wrong_way(sim, vehicle, lane_to_check.getID())
```